### ISSUE TITLE
Sys_error("Invalid argument") -- bin contains msvcp100.dll, but fstar.exe depends on msvcp110.dll

### ISSUE THREAD

**Issue by ForNeVeR** [FStarLang/FStar#866]
While trying to follow examples from the tutorial, I've decided to check them locally. So, I've downloaded Windows version from [the site download section](https://fstar-lang.org/#download), unpacked it to `E:\Programs\fstar` and added `E:\Programs\fstar\bin` to my `PATH`.

So, I have:

```console
PS> fstar --version
F* 0.9.4.0
platform=Windows_x64
compiler=OCaml 4.02.3
date=2017-02-02T19:09:30+05:30
commit=2137ca0 (dirty)
```

After that I've copy-pasted the first tutorial example into `Ex01a.fst` file and now trying to execute the file:

```console
PS> cat .\Ex01a.fst
module Ex01a
open FStar.All
//safe-read-write


type filename = string

(* canWrite is a function specifying whether or not a file f can be written *)
let canWrite (f:filename) =
  match f with
    | "demo/tempfile" -> true
    | _ -> false

(* canRead is also a function ... *)
let canRead (f:filename) =
  canWrite f               (* writeable files are also readable *)
  || f="demo/README"       (* and so is this file *)


val read  : f:filename{canRead f}  -> ML string
let read f  = FStar.IO.print_string ("Dummy read of file " ^ f ^ "\n"); f

val write : f:filename{canWrite f} -> string -> ML unit
let write f s = FStar.IO.print_string ("Dummy write of string " ^ s ^ " to file " ^ f ^ "\n")


let passwd  = "demo/password"
let readme  = "demo/README"
let tmp     = "demo/tempfile"


val staticChecking : unit -> ML unit
let staticChecking () =
  let v1 = read tmp in
  let v2 = read readme in
  (* let v3 = read passwd in -- invalid read, fails type-checking *)
  write tmp "hello!"
  (* ; write passwd "junk" -- invalid write , fails type-checking *)


exception InvalidRead
val checkedRead : filename -> ML string
let checkedRead f =
  if canRead f then read f else raise InvalidRead


assume val checkedWrite : filename -> string -> ML unit

// solution here
//
//


let dynamicChecking () =
  let v1 = checkedRead tmp in
  let v2 = checkedRead readme in
  let v3 = checkedRead passwd in (* this raises exception *)
  checkedWrite tmp "hello!";
  checkedWrite passwd "junk" (* this raises exception *)

let main = staticChecking (); dynamicChecking ()
```

```console
PS> fstar.exe Ex01a.fst
Unexpected error; please file a bug report, ideally with a minimized version of the source program that triggered the error.
Sys_error("Invalid argument")
```

Tried to add `--detail_errors` option with no observable effect.

Obviously I'm doing something wrong, but what exactly? Please help me to figure that out.

**Comment by ForNeVeR** [comment]
It may be important that `z3 --version` exits without any output in my environment:

```console
PS> z3 --version
PS> # nothing
```

**Comment by ForNeVeR** [comment]
A-ha, it was actually the interesting investigation.

I've started to suspect something tricky when I saw `msvcp100.dll` and `msvcr100.dll` in the `fstar/bin` directory. That's obviously remains of some Visual Studio redistributable runtime. "What if — thought I at the moment — the z3 executable have some additional dependencies from the same runtime?" Z3 misbehave looks pretty familiar to anyone who've already dealt with that "DLL hell" on Windows.

So, I've executed the following:
```console
PS> Import-VisualStudioVars 140
PS> dumpbin /dependents E:\Programs\fstar\bin\z3.exe
Microsoft (R) COFF/PE Dumper Version 14.00.24215.1
Copyright (C) Microsoft Corporation.  All rights reserved.


Dump of file E:\Programs\fstar\bin\z3.exe

File Type: EXECUTABLE IMAGE

  Image has the following dependencies:

    MSVCP110.dll
    MSVCR110.dll
    VCOMP110.DLL
    KERNEL32.dll

  Summary

       15000 .data
       58000 .pdata
      229000 .rdata
        D000 .reloc
      85D000 .text
        1000 .tls
```

Look what's interesting: `z3.exe` have dependencies on *three* libraries from VS runtime: there's also `VCOMP110.DLL` that's nowhere to be found.

So, I've downloaded [Visual C++ Redistributable for Visual Studio 2012 Update 4](https://www.microsoft.com/en-us/download/details.aspx?id=30679) from Microsoft, installed that, and now my fstar copy runs without problems.

That means we just have the Windows installer issue: you guys need to put that `VCOMP110.DLL` into the package.

**Comment by s-zanella** [comment]
This is a known issue (see #636). 
Keeping this issue open and closing #636 since there is additional information here.

The solution is to add `msvcp110.dll`, `msvcr110.dll` and `vcomp110.dll` in the release package (and remove `msvcp100.dll` and`msvcr100.dll`). Note that these files are part of the Z3 binary distribution.

**Comment by catalin-hritcu** [comment]
Is this still an issue? Or was it maybe fixed together with issues like #992 (see the discussion there)?

**Comment by ForNeVeR** [comment]
As far as I can see, everything is okay now. Thus closing the issue, thanks for the fix!
